app/cdap/components/NamespaceAdmin/SourceControlManagement/SourceControlManagementForm.tsx

```diff
@@ -0,0 +1,377 @@
+            }}
+            errors={
+              !formState.config?.link && formState.error
+                ? [{ msg: T.translate('commons.requiredFieldMissingMsg').toString() }]
+                : []
+            }
+          />
+          <PropertyRow
+            value={formState.config?.defaultBranch}
+            property={{
+              name: 'defaultBranch',
+              description: T.translate(`${PREFIX}.branch.helperText`).toString(),
+              label: T.translate(`${PREFIX}.branch.label`).toString(),
+            }}
+            onChange={(val) => {
+              handleValueChange(val, 'defaultBranch');
+            }}
+          />
+          <PropertyRow
+            value={formState.config?.pathPrefix}
+            property={{
+              name: 'pathPrefix',
+              description: T.translate(`${PREFIX}.pathPrefix.helperText`).toString(),
+              label: T.translate(`${PREFIX}.pathPrefix.label`).toString(),
+            }}
+            onChange={(val) => {
+              handleValueChange(val, 'pathPrefix');
+            }}
+          />
+        </StyledGroup>
+        <StyledGroup>
+          <BoldHeader>{T.translate(`${PREFIX}.authHeader`)}</BoldHeader>
+          <StyledHr />
+          <PropertyRow
+            value={formState.config?.auth?.type ? formState.config.auth.type : scmAuthType[0].id}
+            property={{
+              name: 'auth',
+              description: T.translate(`${PREFIX}.auth.helperText`).toString(),
+              label: T.translate(`${PREFIX}.auth.label`).toString(),
+              'widget-type': 'radio-group',
+              'widget-attributes': {
+                default: scmAuthType[0].id,
+                layout: 'inline',
+                options: scmAuthType,
+              },
+            }}
+            onChange={(val) => {
+              handleValueChange(val, 'type');
+            }}
+          />
+          {formState.config?.auth?.type === scmAuthType[0].id && (
+            <>
+              <PropertyRow
+                value={formState.config?.auth?.tokenName}
+                property={{
+                  name: 'tokenName',
+                  description: T.translate(`${PREFIX}.auth.pat.tokenNameHelperText`).toString(),
+                  label: T.translate(`${PREFIX}.auth.pat.tokenName`).toString(),
+                  required: true,
+                }}
+                onChange={(val) => {
+                  handleValueChange(val, 'tokenName');
+                }}
+                errors={
+                  !formState.config?.auth?.tokenName && formState.error
+                    ? [{ msg: T.translate('commons.requiredFieldMissingMsg').toString() }]
+                    : []
+                }
+              />
+              <PropertyRow
+                value={formState.config?.auth?.token}
+                property={{
+                  name: 'token',
+                  description: T.translate(`${PREFIX}.auth.pat.tokenHelperText`).toString(),
```

Comment:
`.toString()` shouldn't be required after `T.translate()`

Reply:
I get `Type 'ReactNode' is not assignable to type 'string'.`

Reply:
@njbriggs without .toSTring it creates a react component.

Reply:
Ack, you're right